cluster-autoscaler/cloudprovider/juju/juju_api.go

```diff
@@ -16,7 +17,17 @@ type Unit struct {
 	agent    string
 }
 
+// Note: struct fields must be public in order for unmarshal to
+// correctly populate the data.
+type jujuCloudConfig struct {
+	Username string
```

Comment:
what's the chances i could lobby here for this to be called "user" rather than username?

Reply:
but i get it if it's easier to match the public API

Reply:
the reason I named them as I did was because that's what the [SimpleConnector](https://github.com/juju/juju/blob/fffcf2b1d6cda42866a4811a77e990973f2b2972/api/connector.go#L19) calls them. But I have no preference really, if you think those names align more closely with what the typical juju configurer would expect then we can change it

Reply:
Regarding the uppercase names of the struct, they must be uppercase as go (unfortunately) uses case to denote visibility. The struct fields must be public in order for the yaml package to read the yaml file into the struct, and public fields must start with a capital letter : (